### ISSUE TITLE
Sharedpreference plugin error after flutter upgrade

### ISSUE THREAD

**Issue by vimalmistry** [flutter/flutter#46807]
Working in IOS but not in android. 

I just upgraded flutter sdk and facing some errors related to shared_preferences. I updated this plugin but not working
here is error. I am using latest version which is
`  shared_preferences: ^0.5.6`

## Here is error

` What went wrong:
Could not determine the dependencies of task ':shared_preferences:compileDebugAidl'.
> Could not resolve all task dependencies for configuration ':shared_preferences:debugCompileClasspath'.
   > Could not resolve project :shared_preferences_macos.
     Required by:
         project :shared_preferences
      > Unable to find a matching configuration of project :shared_preferences_macos:
          - None of the consumable configurations have attributes.
   > Could not resolve project :shared_preferences_web.
     Required by:
         project :shared_preferences
      > Unable to find a matching configuration of project :shared_preferences_web:
          - None of the consumable configurations have attributes.
`

## And in android studio

`ERROR: Unable to resolve dependency for ':shared_preferences@debug/compileClasspath': Could not resolve project :shared_preferences_macos.
Show Details
Affected Modules: shared_preferences

ERROR: Unable to resolve dependency for ':shared_preferences@debug/compileClasspath': Could not resolve project :shared_preferences_web.
Show Details
Affected Modules: shared_preferences

ERROR: Unable to resolve dependency for ':shared_preferences@debugAndroidTest/compileClasspath': Could not resolve project :shared_preferences_macos.
Show Details
Affected Modules: shared_preferences

ERROR: Unable to resolve dependency for ':shared_preferences@debugAndroidTest/compileClasspath': Could not resolve project :shared_preferences_web.
Show Details
Affected Modules: shared_preferences

ERROR: Unable to resolve dependency for ':shared_preferences@debugUnitTest/compileClasspath': Could not resolve project :shared_preferences_macos.
Show Details
Affected Modules: shared_preferences `

**Comment by moodstubos** [comment]
same problem:

> Doctor summary (to see all details, run flutter doctor -v):
> [√] Flutter (Channel dev, v1.13.0, on Microsoft Windows [Version 10.0.19041.1], locale de-DE)
>  
> [√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
> [√] Chrome - develop for the web
> [√] Android Studio (version 3.5)
> [√] VS Code (version 1.40.2)
> [√] Connected device (3 available)
> 
> • No issues found!

**Comment by bsr203** [comment]
Similar issue upgrading firebase too
```
Could not determine the dependencies of task ':firebase_core:compileDebugAidl'.
> Could not resolve all task dependencies for configuration ':firebase_core:debugCompileClasspath'.
   > Could not resolve project :firebase_core_web.
     Required by:
         project :firebase_core
      > Unable to find a matching configuration of project :firebase_core_web:
          - None of the consumable configurations have attributes.
```

I added `firebase_core_web` in dependency, still get his error.

**Comment by amirh** [comment]
Can you share more of your log?

unclear if this is also a care of going through the aar build path (like https://github.com/flutter/flutter/issues/46790).

cc @blasten

**Comment by ditman** [comment]
@cyanglaz experienced something **similar** today (in a different gradle step, though):

```
* What went wrong:                                                      
Could not determine the dependencies of task ':app:preDebugBuild'.      
> Could not resolve all task dependencies for configuration ':app:debugRuntimeClasspath'.
   > Could not resolve project :shared_preferences_macos.               
     Required by:                                                       
         project :app > project :shared_preferences                     
      > Unable to find a matching configuration of project :shared_preferences_macos: None of the consumable configurations have attributes.
   > Could not resolve project :shared_preferences_web.                 
     Required by:                                                       
         project :app > project :shared_preferences                     
      > Unable to find a matching configuration of project :shared_preferences_web: None of the consumable configurations have attributes.
                                                                        
* Try:                                                                  
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
                                                                        
* Get more help at https://help.gradle.org       
```

@vimalmistry, @bsr203, @moodstubos can you try the following?

1. Ensure your flutter version is > `1.12.13.hotfix+4` (it probably is)
2. `flutter clean`
3. Delete your `pubsbec.lock`

Then try building again?

The above seemed to solve @cyanglaz's build issues today.

**Comment by edufolly** [comment]
I've clean the ~/.pub-cache folder and executed the project again. Now worked great.

**Comment by The-five-stooges** [comment]
not worked

**Comment by masterfego** [comment]
@edufolly Solution is working for android build. flutter v1.12.13+hotfix.5, on Mac OS X
Locate your flutter SDK folder (mine was not on your location), and delete the .pub-cache dir as suggested.

**Comment by ajaygosh102** [comment]
same problem here:
'''  Could not resolve project :shared_preferences_macos.
     Required by:
         project :shared_preferences
      > Unable to find a matching configuration of project :shared_preferences_macos:
          - None of the consumable configurations have attributes. '''

**Comment by rws08** [comment]
Command to `flutter pub cache repair`

Similar issue to
`video_player, firebase_core, url_launcher`

**Comment by ajaygosh102** [comment]
upgrade to the latest version (v1.12.13+hotfix.6)  worked for me.
``` flutter version v1.12.13+hotfix.6

**Comment by SIRO-Tiger** [comment]
> Command to `flutter pub cache repair`
> 
> Similar issue to
> `video_player, firebase_core, url_launcher`

This seems to solve my problem.

**Comment by yangfanyu** [comment]
> Command to `flutter pub cache repair`
> 
> Similar issue to
> `video_player, firebase_core, url_launcher`

thank you

**Comment by NOA-Charles** [comment]
> Command to `flutter pub cache repair`
> 
> Similar issue to
> `video_player, firebase_core, url_launcher`

Thanks, Helped me a lot.

**Comment by pr-Mais** [comment]
> Command to `flutter pub cache repair`
> 
> Similar issue to
> `video_player, firebase_core, url_launcher`

Worked for me too! thanks a lot.

**Comment by flylai** [comment]
delete `flutter/.pub-cache` and run ` flutter pub get ` again, it worked for me, I use master channel.

**Comment by iapicca** [comment]
possible duplicate https://github.com/flutter/flutter/issues/46767
possible solution  https://github.com/flutter/flutter/issues/46753#issuecomment-564829969

**Comment by leolfsr** [comment]
O comando **flutter pub cache repair**, funcionou para mim.

**Comment by goldenplan** [comment]
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':shared_preferences:compileDebugAidl'.
> Could not resolve all task dependencies for configuration ':shared_preferences:debugCompileClasspath'.
   > Could not resolve project :shared_preferences_macos.
     Required by:
         project :shared_preferences
      > Unable to find a matching configuration of project :shared_preferences_macos:
          - None of the consumable configurations have attributes.
   > Could not resolve project :shared_preferences_web.
     Required by:
         project :shared_preferences
      > Unable to find a matching configuration of project :shared_preferences_web:
          - None of the consumable configurations have attributes.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 6s
Finished with error: Gradle task assembleDebug failed with exit code 1

**Comment by goldenplan** [comment]
this error only on flutter dev channel

**Comment by amirh** [comment]
Did anyone experience this when upgrading from flutter stable to the new flutter stable release? (we're trying to understand if this problem occurs only on the dev channel per https://github.com/flutter/flutter/issues/46898#issuecomment-565179185 )

**Comment by alexda12** [comment]
This error has happened to me on **Flutter Stable 1.12.13 hotfix 5**.

The issue occurs when you reference plugins such as : 

google_sign_in: 4.1.0
firebase_auth: 0.15.2
cloud_firestore: 0.13.0+1
cloud_functions: 0.4.1+5
shared_preferences: 0.5.6

etc  which support web by default at those versions.

I've actually managed to resolve this by : 

**Either :** 

- downgrading those web by default versions (also skip past MAC OS versions)

**OR (Better)** 

1. Going to flutter->.pub-cache and remove everything in this folder
2. Remove entire .gradle folder
3. Ensure your gradle.properties has this version of the URL :  

**distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.2-all.zip**

4. Ensure your build.gradle contains **compileSdkVersion of 28** (I also changed targetSdkVersion to 28)

5. Use   classpath 'com.android.tools.build:gradle:3.5.0' in android/build.gradle 

**Once all that is done, do :**  

1. Flutter clean
2. Make a change to pubspec.yaml and SAVE file
3. Flutter pub cache repair

You should see **output** console that MAC_OS/ WEB etc plugins are ALSO being downloaded which will then resolve the **could not resolve project** issues

**Comment by amirh** [comment]
shared_preferences_web and shared_preferences_macos versions with a workaround have been published. `flutter pub upgrade` should fetch the newer versions and resolve the issue (if someone that experienced the issue can confirm that it is resolve after upgrading it will be helpful).

Note that there are still other plugins waiting to be published with the workaround, full status is at: https://github.com/flutter/flutter/issues/46898#issuecomment-565132887

**Comment by amirh** [comment]
@alexda12 can you confirm whether before upgrading to 1.2.13+hotfix.5 you've been on the dev channel or on the stable channel?

**Comment by alexda12** [comment]
@amirh Prior to the upgrade on **Friday**  I was on **Stable v1.9.1+hotfix.6**

**Comment by amirh** [comment]
Thanks! @blasten FYI

**Comment by Arcturus5404** [comment]
Also had an issue with shared_preferences 0.5.6, downgrading to the previous fixed it for me; for now.. After upgrading flutter, I also had these problems:

```
Android dependency 'androidx.lifecycle:lifecycle-runtime' has different version for the compile (2.0.0-rc01) and runtime (2.0.0) classpath. You should manually set the same version via DependencyResolution
```

Fixed those with this in the build.gradle:

```
            if (details.requested.group == 'androidx.lifecycle'
                    && !details.requested.name.contains('multidex') ) {
                details.useVersion "2.0.0"
            }
            if (details.requested.group == 'androidx.versionedparcelable'
                    && !details.requested.name.contains('multidex') ) {
                details.useVersion "1.0.0"
            }
            if (details.requested.group == 'androidx.core'
                    && !details.requested.name.contains('multidex') ) {
                details.useVersion "1.0.2"
            }
```

All done with the stable channel btw.

**Comment by humazed** [comment]
Instead of `flutter pub cache repair` and redownloading the whole cheche just deleted the offending plugin.

go to `.pub-cache\hosted\pub.dartlang.org` and sort by modification date and delete the Sharedpreference plugin

**Comment by leolfsr** [comment]
Segui esse passo a passo, resolveu. 
[https://dev.to/pedromassango/corrigindo-o-erro-androidx-no-flutter-5eeg](url)

**Comment by amirh** [comment]
The issue should now be resolved, if you're hitting the failure please run `flutter pub upgrade` to upgrade to the latest version of all packages.

If you're still hitting the issue after upgrading your project's dependencies please leave a comment on: https://github.com/flutter/flutter/issues/46898

**Comment by blasten** [comment]
Could anyone confirm that you opened Android Studio after upgrading to the latest Flutter stable?  This seems to be the only case we didn't catch in our testing.

**Comment by sunvr06** [comment]
> Command to `flutter pub cache repair`
> 
> Similar issue to
> `video_player, firebase_core, url_launcher`

thank you bro

**Comment by kuhnroyal** [comment]
I had to delete the `pubspec.lock` to fix this.
The issue started yesterday after upgrading flutter from `v1.12.13+hotfix.5` to `hotfix.6`.
No changes to the `pubspec.yaml` - it was on `shared_preferences: 0.5.6` the whole time.

After the `pubspec.lock` is regenerated it contains diffs with newer `shared_preferences_macos` and `shared_preferences_web` dependencies.

https://github.com/flutter/plugins/blob/master/packages/shared_preferences/shared_preferences/pubspec.yaml#L30 
```
  shared_preferences_macos: ^0.0.1
  shared_preferences_web: ^0.1.2
```
These are still wrong as the min specified version does't exist or doesn't work. If you already have them in the `pubspec.lock`, it won't update.

One way to solve this, is probably running `flutter pub upgrade` but a lot of people don't use that command because of https://github.com/flutter/flutter/issues/12627.

So please just release a shared_preferences 0.5.7 with the fixed min requirements that actually work.
```
  shared_preferences_macos: ^0.0.1+1
  shared_preferences_web: ^0.1.2+1
```

**Comment by nareshgnT** [comment]
I get this issue only when change some configuration like launch screen under android folder, when it builds it create settings_aar.gradle and try to compile the dependencies. During that i get gradle version issue

**Comment by Stinox22** [comment]
Updating the gradle wrapper to 5.4.1 helped me.

**Comment by AlexStefan** [comment]
I'm getting into the same error as I've upgraded to the latest version of Flutter.

`Because project depends on shared_preferences >=0.5.6 which requires Flutter SDK version >=1.12.13+hotfix.4 <2.0.0, version solving failed.`

Tried already to delete the lock file, clean, pub upgrade, pub cache repair, but none of them worked.

**Comment by iapicca** [comment]
Could everyone who still has this problem please file a new issue with the exact descriptions what happens, logs and the output of 'flutter doctor -v'  please.
All system setups can be slightly different so it's always better to open new issues and reference related issues.
